### PR TITLE
Basic setup

### PR DESCRIPTION
**Pull request by sergiitk**
## Basic setup
- Npm init
- Add minimal nodejs requirement
- Basic NorthstarClient class
- Integrate simple mocha tests
- Enable linting via eslint based on [@dosomething/eslint-config](/DoSomething/eslint-config)
- Setup linting

### CODE REVIEW

**Comment by sergiitk** [comment]
CC @DFurnes: please take a peek if you have a second.

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+  extends: "@dosomething/eslint-config"
+  parserOptions:
+    # Node requires 'use strict' to use built-in ES6 compatibility.
+    # Use script source type, so lint doesn't throw error:
+    # 'use strict' is unnecessary inside of modules.
+    sourceType: "script"
+  rules:
+    # We don't use babel here, so require 'use global'.
```
Comment (DFurnes) [inline .eslintrc]: Just to avoid confusion, this should be `use strict` rather than `use global`.

**Comment by DFurnes** [comment]
Looks good! If it's not too late, it might be nice to rename this `northstar-js` under the assumption that it could work in the browser too. Could be useful to have for client-side applications. 😄

**Comment by sergiitk** [comment]
Thanks! I intended it as a NodeJs app though, using its native ES6 support. But yes, it makes sense to rename it anyways.
What would you prefer `northstar-js` ot `northstar-js-client`?

**Comment by DFurnes** [comment]
I think `northstar-js` makes sense to keep consistency with [`northstar-php`](https://github.com/DoSomething/northstar-php).

**Comment by sergiitk** [comment]
Heh. I've seen it. Honestly, I would rather rename `northstar-php` to `northstar-php-client`. Is there a guideline we refer to for naming repos?

**Comment by DFurnes** [comment]
I don't think we have any established rules. I'd prefer not to rename `northstar-php` since we'd also have to update references anywhere else its used, but you're certainly free to name this package as you please.

**Comment by sergiitk** [comment]
Ok, I'll rename it to `northstar-js` when this PR is merged.

File: .eslintrc (hunk 1 above)
Comment (sergiitk) [inline .eslintrc]: Right, thank you. Good 👀!

**Comment by deezone** [comment]
👍

**Pull request merged**